core/consensus/babe/babe_synchronizer.hpp

```diff
@@ -3,42 +3,36 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifndef KAGOME_CORE_CONSENSUS_BABE_BABE_SYNCHRONIZER_HPP
-#define KAGOME_CORE_CONSENSUS_BABE_BABE_SYNCHRONIZER_HPP
+#ifndef KAGOME_CONSENSUS_BABE_SYNCHRONIZER
+#define KAGOME_CONSENSUS_BABE_SYNCHRONIZER
 
 #include <libp2p/peer/peer_id.hpp>
 
-#include "primitives/authority.hpp"
-#include "primitives/block.hpp"
-#include "primitives/block_data.hpp"
-#include "primitives/block_id.hpp"
+#include "outcome/outcome.hpp"
+#include "primitives/block_header.hpp"
+#include "primitives/common.hpp"
 
 namespace kagome::consensus {
 
-  /**
-   * @brief Iterates over the list of accessible peers and tries to fetch
-   * missing blocks from them
-   */
   class BabeSynchronizer {
    public:
-    using BlocksHandler =
-        std::function<void(boost::optional<std::reference_wrapper<
-                               const std::vector<primitives::BlockData>>>)>;
+    using SyncResultHandler =
+        std::function<void(outcome::result<primitives::BlockInfo>)>;
 
     virtual ~BabeSynchronizer() = default;
 
     /**
-     * Request blocks between provided ones
-     * @param from block id of the first requested block
-     * @param to block hash of the last requested block
-     * @param block_list_handler handles received blocks
+     * Enqueues new-noticed block for loading
      */
-    virtual void request(const primitives::BlockId &from,
-                         const primitives::BlockHash &to,
+    virtual void enqueue(const primitives::BlockInfo &block_info,
```

Comment:
I believe it would improve readability if the method name were more detailed, to abandon the need to switch to its documentation to understand what it's doing. Also, handler is a bit cryptic without reading the method's implementation, and even when reading it it's not trivial to fully comprehend when it's called and what exactly it's supposed to handle, given that it's mentioned in block_synchronizer_impl.cpp 48 times, according to CLion. Renaming it or adding additional status arguments to it might help.

Reply:
Renamed to two different name and commented